src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -18468,6 +18479,400 @@ public void Main()
+                // (18,9): warning CS8629: Nullable value type may be null.
+                //         new COpen<T?>().P1.Value.ToString(); // 5
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "new COpen<T?>().P1").WithLocation(18, 9),
+                // (19,9): warning CS8629: Nullable value type may be null.
+                //         new CStruct<T>().P5.Value.ToString(); // 6
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "new CStruct<T>().P5").WithLocation(19, 9)
+            };
+
+            var comp = CreateNullableCompilation(source, references: new[] { lib.EmitToImageReference() });
+            comp.VerifyDiagnostics(expected);
+
+            var comp2 = CreateNullableCompilation(new[] { source, lib_cs, MaybeNullAttributeDefinition });
+            comp2.VerifyDiagnostics(expected);
+        }
+
+        [Fact]
+        public void MaybeNull_Property_OnVirtualProperty()
+        {
+            // Consider warning on improper nullability attributes in OHI? https://github.com/dotnet/roslyn/issues/36039
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+public class Base
+{
+    [MaybeNull] public virtual string P => throw null!;
+    public virtual string P2 => throw null!;
+}
+public class C : Base
+{
+    public override string P => throw null!;
+    [MaybeNull] public override string P2 => throw null!;
+
+    static void M(C c, Base b)
+    {
+        b.P.ToString(); // 1
+        c.P.ToString();
+
+        b.P2.ToString();
+        c.P2.ToString(); // 2
+    }
+}";
+            var comp = CreateNullableCompilation(new[] { source, MaybeNullAttributeDefinition });
+            comp.VerifyDiagnostics(
+                // (15,9): warning CS8602: Dereference of a possibly null reference.
+                //         b.P.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "b.P").WithLocation(15, 9),
+                // (19,9): warning CS8602: Dereference of a possibly null reference.
+                //         c.P2.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c.P2").WithLocation(19, 9)
+                );
+        }
+
+        [Fact]
+        public void MaybeNull_Property_OnVirtualProperty_OnlyOverridingSetter()
+        {
+            // Consider warning on improper nullability attributes in OHI? https://github.com/dotnet/roslyn/issues/36039
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+public class Base
+{
+    [MaybeNull] public virtual string P { get { throw null!; } set { throw null!; } }
+    public virtual string P2 { get { throw null!; } set { throw null!; } }
+}
+public class C : Base
+{
+    public override string P { set { throw null!; } }
+    [MaybeNull] public override string P2 { set { throw null!; } }
+
+    static void M(C c, Base b)
+    {
+        b.P.ToString(); // 1
+        c.P.ToString();
```

Comment:
>c.P.ToString(); [](start = 8, length = 15)

A warning should be on this line

Reply:
It looks like we are missing tests for usage of a setter when only getter is overridden. Do we have tests for both accessors when both are overridden?

---
In reply to: [295461748](https://github.com/dotnet/roslyn/pull/36444#discussion_r295461748) [](ancestors = 295461748)

Reply:
Also need to have tests for overriding for indexers, including attributes applied to explictly declared parameters.

---
In reply to: [295470186](https://github.com/dotnet/roslyn/pull/36444#discussion_r295470186) [](ancestors = 295470186,295461748)